app/decorators/oh_admin/account_chart.rb

```diff
@@ -0,0 +1,57 @@
+class OhAdmin::AccountChart
+  def initialize(period)
+    @period = period
+    process_account_data(period)
+  end
+
+  def render
+    chart = ACCOUNTS_CHART_DEFAULTS
+    chart['series'][0][:data] = @spam.values
+    chart['series'][1][:data] = @regular.values
+    chart['series'][2][:data] = @total_count
+    chart['xAxis']['tickInterval'] = set_interval(@period)
+    chart['xAxis']['categories'] = @x_axis
+    chart.to_json
+  end
+
+  private
+
+  def process_account_data(period)
+    from = period.months.ago.to_date
+    yesterday = Date.yesterday
+    @spam = spam_accounts(from, yesterday)
+    @regular = regular_accounts(from, yesterday)
+    @total_count = []
+    @x_axis = []
+    fill_zero_gaps(from, yesterday)
+  end
+
+  def fill_zero_gaps(from, yesterday)
+    (from..yesterday).each do |date|
+      @spam[date] = 0 if @spam[date].nil?
+      @regular[date] = 0 if @regular[date].nil?
+      @total_count <<  total_accounts(date)
+      @x_axis << date.strftime('%a, %b %d')
+    end
+  end
+
+  def spam_accounts(from, to)
+    Account.group('date(created_at)').where(created_at: from..to, level: Account::Access::SPAM).count
+  end
+
+  def regular_accounts(from, to)
+    Account.group('date(created_at)').where(created_at: from..to, level: Account::Access::DEFAULT).count
+  end
+
+  def total_accounts(date)
+    Account.where('DATE(created_at) <= ?', date).where(level: Account::Access::DEFAULT).count
```

Comment:
This is a COSTLY and DEADLY query (and loops through 90, 180 and 360 iterations) and needs help on it